### DISCUSSION TITLE
Does Github Advanced Security detect code smells

### DISCUSSION THREAD

**Discussion by asyedcloud** [community/community#130985] [Code Security] [answered]
### Select Topic Area

Question

### Body

Does Github Advanced Security detect code smell? I'm running a trial of this product with view to replace sonarqube but it doesnt pickup any code smells. Do you know why? I have used all of the following queries

code-scanning
security-experimental
security-extended
security-and-quality

sonarqube picks up 10s of code smell in the same repo that im trialling with github advanced security

**Comment by SajanGhimire1** [comment, marked as answer]
Hi asyedcloud
GitHub Advanced Security focuses primarily on identifying security vulnerabilities such as exposed secrets and potential security flaws rather than detecting code smells. Code smells, which are indicators of potential deeper problems in your code that may not necessarily lead to immediate bugs but could suggest improvements for better code maintainability and readability, are not within the scope of what GitHub Advanced Security is designed to detect.

For identifying and managing code smells, tools like SonarQube are specifically tailored to analyze code quality, including detecting code smells across various programming languages. They offer detailed insights and suggestions on how to improve code health, which goes beyond what GitHub Advanced Security is intended for.

If your primary need is to manage code quality, including spotting and fixing code smells, continuing with or integrating tools like SonarQube in your workflow would be advisable. GitHub Advanced Security is excellent for enhancing security through features like code scanning, secret scanning, and dependency review but does not replace the code quality analysis tools that are focused on structural code problems and smells.
